File: submissions/1354-find-players-with-zero-or-one-losses/solution.ts
function findWinners(matches: number[][]): number[][] {
   const players = new Map();
   let losers = [];
   let winners = [];
   for (let i = 0; i < matches.length; i++) {
       const loser = matches[i][1]
       const winner = matches[i][0]

       players.set(loser, (players.get(loser) + 1 || 1));
       players.set(winner, (players.get(winner) && players.get(winner) !== 0 ? players.get(winner) : 0));

   } 
    for (const [key, value] of players){
        if (value === 1){
            losers.push(key);
        } else if(value === 0){
            winners.push(key);
        }
    }
    return [winners.sort((a,b)=> a - b), losers.sort((a,b)=> a - b)]
};

File: submissions/1354-find-players-with-zero-or-one-losses/solution.py
class Solution:
    def findWinners(self, matches: List[List[int]]) -> List[List[int]]:
        loss_counts = defaultdict(int)
        
        for winner, loser in matches:
            loss_counts[winner] 
            loss_counts[loser] += 1
        
        no_loss = [player for player, losses in loss_counts.items() if losses == 0]
        one_loss = [player for player, losses in loss_counts.items() if losses == 1]
        
        return [sorted(no_loss), sorted(one_loss)]
